### PR TITLE
Add more specific phpdoc

### PR DESCRIPTION
**Pull request by core23**
## Subject

Add more specific phpdoc

I am targeting this branch, because this is BC.

### CODE REVIEW

File: src/Admin/AdminInterface.php [hunk 1]
```diff
@@ -484,7 +484,7 @@ public function getTranslationDomain();
     /**
      * Return the form groups.
      *
-     * @return array
+     * @return array|false
```
Comment (phansys) [inline src/Admin/AdminInterface.php:487]: We could explain which cases makes this method to return `false`.

Comment (core23) [reply src/Admin/AdminInterface.php]: Fixed

File: src/Admin/AdminInterface.php [hunk 2]
```diff
@@ -484,7 +484,7 @@ public function getTranslationDomain();
     /**
      * Return the form groups.
      *
-     * @return array
+     * @return array<string, mixed>|false (false if the groups have not been initilized)
```
Comment (VincentLanglet) [inline src/Admin/AdminInterface.php:487]: initialized

File: src/Admin/AdminInterface.php [hunk 3]
```diff
@@ -511,7 +511,7 @@ public function removeFieldFromFormGroup($key);
     /**
      * Returns the show groups.
      *
-     * @return array
+     * @return array<string, mixed>|false (false if the groups have not been initilized)
```
Comment (VincentLanglet) [inline src/Admin/AdminInterface.php:514]: initialized

**Comment by greg0ire** [comment]
Thanks @core23

**Pull request merged**